### PR TITLE
Add Pandas extension array test suites for TensorArray

### PR DESCRIPTION
**Pull request by BryanCutler**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,4 +5,4 @@ regex
 fastparquet
 # TODO: The following dependency should go away when we switch to Python 3.8.
 memoized-property
-
+pytest
```
Comment (BryanCutler) [inline requirements.txt:8]: Pandas uses fixtures which requires pytest

**Comment by BryanCutler** [comment]
@frreiss , I have added all the test suites for TensorArraybut there got to be  too many errors, so I'll have to work through them one by one and see what we can support. The other extension types should be easier I believe. Adding these tests does require pytest because of the use of fixtures, so wanted to make sure you're ok with that.

**Comment by frreiss** [comment]
@BryanCutler sure, adding `pytest` is fine.

File: text_extensions_for_pandas/array/test_tensor.py [hunk 2]
```diff
@@ -299,11 +313,14 @@ def test_sum(self):
...
-                b    [[1, 1], [1, 1]]
-                c    [[3, 3], [3, 3]]"""
+                             value
+                key               
+                a   [[2 2]
+                 [2 2]]
+                b   [[1 1]
+                 [1 1]]
+                c   [[3 3]
+                 [3 3]]"""
```
Comment (BryanCutler) [inline text_extensions_for_pandas/array/test_tensor.py:323]: The change in `_from_sequence` caused this. This looks worse, but `_from_sequence` was wrong before so we can figure out how to format this better later.

**Comment by BryanCutler** [comment]
@frreiss are you ok with merging this as is and I can start working on adding tests for the other extension arrays?

File: text_extensions_for_pandas/array/test_tensor.py (hunk 2 above)
Comment (frreiss) [reply text_extensions_for_pandas/array/test_tensor.py:323]: ok

**Comment by frreiss** [comment]
Sure, this looks OK to merge.

**Comment by BryanCutler** [comment]
Thanks @frreiss !

**Pull request merged**